CMake should check for minimum GCC version requirement for C++14 support

CoreNEURON currently requires compiler with C++14 support. We have following in top level CMake:

```
set(CMAKE_CXX_STANDARD 14)
set(CMAKE_CXX_STANDARD_REQUIRED ON)
```

This make sure to add C++14 flags (e.g. -std=c++1y) but this is not sufficient for C++14 and result into following error:

```
[  6%] Building CXX object external/coreneuron/coreneuron/CMakeFiles/coreneuron.dir/nrniv/cvodestb.cpp.o
In file included from /home/ext_salvadordura_gmail_com/nrn/external/coreneuron/coreneuron/nrniv/cvodestb.cpp:29:0:
/home/ext_salvadordura_gmail_com/nrn/external/coreneuron/coreneuron/coreneuron.hpp:77:7: internal compiler error: in gen_type_die_with_usage, at dwarf2out.c:19525
 class CoreNeuron {
       ^
Please submit a full bug report,
with preprocessed source if appropriate.
See <http://bugzilla.redhat.com/bugzilla> for instructions.
Preprocessed source stored into /tmp/ccqWAT4G.out file, please attach this to your bugreport.
make[2]: *** [external/coreneuron/coreneuron/CMakeFiles/coreneuron.dir/nrniv/cvodestb.cpp.o] Error 1
make[1]: *** [external/coreneuron/coreneuron/CMakeFiles/coreneuron.dir/all] Error 2
make: *** [all] Error 2
```

See #219 ([comment](https://github.com/BlueBrain/CoreNeuron/issues/219#issuecomment-568796069)).

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,8 +24,10 @@ set(CMAKE_BUILD_TYPE
     RelWithDebInfo
     CACHE STRING "Empty or one of Debug, Release, RelWithDebInfo")
 
-if (NOT "cxx_std_14" IN_LIST CMAKE_CXX_COMPILE_FEATURES)
-  message(FATAL_ERROR "This compiler does not fully support C++14, choose an higher version or an other compiler.")
+if(NOT "cxx_std_14" IN_LIST CMAKE_CXX_COMPILE_FEATURES)
+  message(
+    FATAL_ERROR
+      "This compiler does not fully support C++14, choose an higher version or an other compiler.")
 endif()
 
 # =============================================================================

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,6 +24,10 @@ set(CMAKE_BUILD_TYPE
     RelWithDebInfo
     CACHE STRING "Empty or one of Debug, Release, RelWithDebInfo")
 
+if (NOT "cxx_std_14" IN_LIST CMAKE_CXX_COMPILE_FEATURES)
+  message(FATAL_ERROR "This compiler does not fully support C++14, choose an higher version or an other compiler.")
+endif()
+
 # =============================================================================
 # Settings to enable project as submodule
 # =============================================================================

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,7 +24,8 @@ set(CMAKE_BUILD_TYPE
     RelWithDebInfo
     CACHE STRING "Empty or one of Debug, Release, RelWithDebInfo")
 
-if(NOT "cxx_std_14" IN_LIST CMAKE_CXX_COMPILE_FEATURES OR NOT "cxx_digit_separators" IN_LIST CMAKE_CXX_COMPILE_FEATURES)
+if(NOT "cxx_std_14" IN_LIST CMAKE_CXX_COMPILE_FEATURES OR NOT "cxx_digit_separators" IN_LIST
+                                                          CMAKE_CXX_COMPILE_FEATURES)
   message(
     FATAL_ERROR
       "This compiler does not fully support C++14, choose an higher version or an other compiler.")

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,7 +24,7 @@ set(CMAKE_BUILD_TYPE
     RelWithDebInfo
     CACHE STRING "Empty or one of Debug, Release, RelWithDebInfo")
 
-if(NOT "cxx_std_14" IN_LIST CMAKE_CXX_COMPILE_FEATURES)
+if(NOT "cxx_std_14" IN_LIST CMAKE_CXX_COMPILE_FEATURES OR NOT "cxx_digit_separators" IN_LIST CMAKE_CXX_COMPILE_FEATURES)
   message(
     FATAL_ERROR
       "This compiler does not fully support C++14, choose an higher version or an other compiler.")

